src/Altinn.Apps/AppTemplates/AspNet/App.IntegrationTests/Utils/SetupUtil.cs

```diff
@@ -52,17 +52,20 @@ public static HttpClient GetTestClient(CustomWebApplicationFactory<Startup> cust
                 builder.ConfigureTestServices(services =>
                 {
                     services.Configure<AppSettings>(appSettingSection);
-                    services.AddSingleton<IInstance, InstanceMockSI>();
-                    services.AddSingleton<IData, DataMockSI>();
-                    services.AddSingleton<IInstanceEvent, InstanceEventAppSIMock>();
-                    services.AddSingleton<IDSF, DSFMockSI>();
-                    services.AddSingleton<IER, ERMockSI>();
-                    services.AddSingleton<IRegister, RegisterMockSI>();
+
                     services.AddSingleton<Altinn.Common.PEP.Interfaces.IPDP, PepWithPDPAuthorizationMockSI>();
-                    services.AddSingleton<IApplication, ApplicationMockSI>();
-                    services.AddTransient<IProfile, ProfileMockSI>();
+
                     services.AddSingleton<IValidation, ValidationAppSI>();
-                    services.AddSingleton<IPDF, PDFMockSI>();
+
+                    services.AddHttpClient<IApplication, ApplicationMockSI>();
```

Comment:
Do we need the mocks to have HttpClients? Are we testing Dependency Injection?

Reply:
@SandGrainOne after your comment i tested to see if wee need to override httpclients as httpclients. And the answer is no. We can override httpclients as transients services. Fixed